### PR TITLE
Dust and lenderActions changes

### PR DESCRIPTION
**Pull request by ith-harvey**
## Description
* Created `testRemoveQuoteDust` and `testMoveQuoteDust` tests
* added dust limit checks in the following functions (ensuring that the bucket) is not left with deposit after the action):
  * `removeMaxDeposit()`
  * `addQuoteToken()`
  * `moveQuoteToken()`
 
## Purpose
* Resolves `Kirill L-01` (https://github.com/k1rill-fedoseev/audits/blob/master/solo/Ajna.md)

## Impact

old
```
addQuoteToken                        | 1950            | 252735 | 169787 | 723088 | 944     |
moveQuoteToken                       | 1918            | 149390 | 107324 | 634633 | 41      |
```

new
```
addQuoteToken                        | 1950            | 253720 | 170124 | 723195 | 949     |
moveQuoteToken                       | 1918            | 146633 | 107386 | 634829 | 44      |
```

### size
old
```
  ERC721Pool               -  24,069B  (97.93%)
  ERC20Pool                -  23,360B  (95.05%)
  PositionManager          -  17,801B  (72.43%)
  TakerActions             -  15,866B  (64.56%)
  PositionNFTSVG           -  15,271B  (62.14%)
  PoolInfoUtils            -  14,399B  (58.59%)
  LenderActions            -  10,715B  (43.60%)
  BorrowerActions          -   9,358B  (38.08%)
  SettlerActions           -   9,349B  (38.04%)
  KickerActions            -   9,153B  (37.24%)
  PoolCommons              -   9,089B  (36.98%)
  RewardsManager           -   8,693B  (35.37%)
  PoolInfoUtilsMulticall   -   8,539B  (34.74%)
  ERC721                   -   4,276B  (17.40%)
  ERC721PoolFactory        -   3,742B  (15.23%)
  LPActions                -   2,952B  (12.01%)
  ERC20PoolFactory         -   2,564B  (10.43%)
```

new
```
  ERC721Pool               -  24,069B  (97.93%)
  ERC20Pool                -  23,360B  (95.05%)
  PositionManager          -  17,801B  (72.43%)
  TakerActions             -  15,866B  (64.56%)
  PositionNFTSVG           -  15,271B  (62.14%)
  PoolInfoUtils            -  14,399B  (58.59%)
  LenderActions            -  10,823B  (44.04%)
  BorrowerActions          -   9,358B  (38.08%)
  SettlerActions           -   9,349B  (38.04%)
  KickerActions            -   9,153B  (37.24%)
  PoolCommons              -   9,089B  (36.98%)
  RewardsManager           -   8,693B  (35.37%)
  PoolInfoUtilsMulticall   -   8,539B  (34.74%)
  ERC721                   -   4,276B  (17.40%)
  ERC721PoolFactory        -   3,742B  (15.23%)
  LPActions                -   2,952B  (12.01%)
  ERC20PoolFactory         -   2,564B  (10.43%)
  ERC20                    -   2,114B  (8.60%)
```
## Tasks

- [x] Changes to protocol contracts are covered by unit tests executed by CI.
- [x] Protocol contract size limits have not been exceeded.
- [x] Gas consumption for impacted transactions have been compared with the target branch, and nontrivial changes cited in the _Impact_ section above.
- [x] Scope labels have been assigned as appropriate.
- [x] Invariant tests have been manually executed as appropriate for the nature of the change.

### CODE REVIEW

**Comment by ith-harvey** [comment]
Copped over changes from -> `kirill-L01-dust-test`

**Review by EdNoepel — APPROVED** [review]
LGTM

**Review by prat-gpt — APPROVED** [review]
LGTM

File: src/libraries/external/LenderActions.sol [hunk 1]
```diff
@@ -841,4 +842,4 @@ library LenderActions {
         // update FenwickTree
         Deposits.unscaledRemove(deposits_, params_.index, unscaledRemovedAmount);
     }
-}
+}
```
Comment (MikeHathaway) [inline src/libraries/external/LenderActions.sol:845]: nit: new line at EOF

**Review by MikeHathaway — APPROVED** [review]
LGTM

**Pull request merged**